src/mds/MDCache.cc

```diff
@@ -8465,7 +8465,11 @@ int MDCache::path_traverse(MDRequestRef& mdr, MDSContextFactory& cf,
 	  // force client to flush async dir operation if necessary
 	  if (cur->filelock.is_cached())
 	    lov.add_wrlock(&cur->filelock);
-	  lov.add_rdlock(&dn->lock);
+	  if (mdr->is_xlocked(&dn->lock)) {
+	    lov.add_xlock(&dn->lock);
+	  } else {
+	    lov.add_rdlock(&dn->lock);
+	  }
```

Comment:
Since cf5fdcd, `openc` will not xlock an existing non-null dentry if `O_EXCL` is not set. Instead it will acquire a rdlock:

https://github.com/ceph/ceph/commit/cf5fdcd6e2d22f35b670d3c13ae8b7c9b9d2616d#diff-277dc6e796ccecb6aa14c9357f7a86898d6ddcf4113c110a4e00e0a10f6fefa7R4476

So I don't think this commit will do anything?

Reply:
I think you are right.

Because only when `dnl->is_null() || !want_inode` will `true` will the `_openc()` `xlock` the dentry. While if `dnl->is_null()` is true it should go to `_open()`. And the `rdlock_path_xlock_dentry()` in `_openc()` will always make `want_inode` to `true`. So the `_openc()` won't `xlock` the dentry.